src/GraphQL/Types/Collections/SchemaTypes.cs

```diff
@@ -501,6 +535,53 @@ private void AddTypeIfNotRegistered(IGraphType type, TypeCollectionContext conte
             }
         }
 
+        private object RebuildType(Type type, bool input, List<(Type, Type)> typeMappings)
```

Comment:
This looks like an ugly hack -- returning an error as a string or a value as a `Type`.  How about returning an error as a string, or null if no error, with an out variable for the type.  Or return a bool with two out variables.  Or return a tuple with 2/3 variables.  But not an object that could be either a `string` or a `Type`.

Reply:
Believe me, I did it first. It turns out to be even worse.

Reply:
Well...at least it's private.  Just feels wrong.